### ISSUE TITLE
Geographical hierarchies

### ISSUE THREAD

**Issue by BillSwirrl** [OpenGovIntelligence/json-qb#10]
This could actually apply more generally, but a particularly common use case is wanting to select data for a collection of areas - often 'all areas of type X that are within some other area Y'

eg 'all council areas in Scotland'

or

'all data zones in Edinburgh'

it would be useful to offer a facility like this in the API, but will have to think about how to make it suitably generic.  It would come with some requirements on how the data was represented in the underlying store

**Comment by zeginis** [comment]
This could by implemented through a filtering mechanism for hierarchical data. This requires the hierarchical data to be well formated e.g.
- Define the hierarchical levels e.g. council area, data zone
- Define the members of each hierarchical level e.g. council area:{ area1, area2 ...},  data zone:{zone1, zone2,...}
- Define the hierarchical relations e.g. area1-> Scotland,  area2-> Wales,  zone1 -> Edinburgh, zone2-> London ... etc

**Comment by BillSwirrl** [comment]
Yes, we have implemented a simple and flexible thing like that for data in PublishMyData - an area can be a 'memberOf' one or more collections of areas, and an area can be 'within' any number of larger areas.  It works well and gives you a lot of freedom to decide what kind of collections and what kind of hierarchical relationships are useful.  

It doesn't require a strict 'tree' style hierarchy.

For example see:

http://statistics.gov.scot/doc/statistical-geography/S01009467 (have a look at the API tab)

That idea could be generalised to work in any RDF data cube collection.  We could perhaps agree a simple ontology that isn't associated specifically with PublishMyData
